ibm/service/vpc/resource_ibm_is_subnet.go

```diff
@@ -613,6 +635,14 @@ func subnetUpdate(d *schema.ResourceData, meta interface{}, id string) error {
 			return err
 		}*/
 	}
+	if d.HasChange(isSubnetRoutingTableCrn) {
```

Comment:
please verify if this handles the case where crn is removed and if is added and vice versa

Reply:
resource "ibm_is_subnet" "testacc_subnet" {
  name            = "terrraform-test"
  vpc             = ibm_is_vpc.testacc_vpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
  # routing_table = ibm_is_vpc.testacc_vpc.default_routing_table
  # routing_table_crn = ibm_is_vpc.testacc_vpc.default_routing_table_crn
  routing_table_crn = ibm_is_vpc_routing_table.test_route_rt.crn
  # routing_table = ibm_is_vpc_routing_table.test_route_rt.routing_table
}

same routing table/crn no diff, different id or crn will result into force replace.

<img width="1099" alt="Screenshot 2024-08-04 at 6 42 57 PM" src="https://github.com/user-attachments/assets/6ddc671e-60e9-4622-aeb7-7531852a3d02">